src/pages/rss.xml.js

```diff
@@ -2,12 +2,12 @@ import rss from "@astrojs/rss";
 import { getCollection } from "astro:content";
 
 export async function GET() {
-  const posts = await getCollection('posts');
+  const posts = await getCollection("blog");
   return rss({
     title: 'Astro Learner | Blog',
     description: 'My journey learning Astro',
     site: 'https://my-blog-site.netlify.app',
-    items: posts.map((post) => ({
+    items: posts.map((post:any) => ({
```

Comment:
It's a `.js` file, so we can't use a type annotation here:
```suggestion
    items: posts.map((post) => ({
```